core/sodasql/soda_server_client/soda_server_client.py

```diff
@@ -42,41 +42,75 @@ def __init__(self,
+            The origin.
+
+        Returns
+        -------
+        out : dict
+            The response.
+        """
+        scan_yml_columns = scan_yml_columns or dict()
         soda_column_cfgs = {}
-        if scan_yml.columns:
-            for column_name in scan_yml.columns:
-                scan_yml_column: ScanYmlColumn = scan_yml.columns[column_name]
-                soda_column_cfg = {}
-                if scan_yml_column.missing:
-                    if scan_yml_column.missing.values:
-                        soda_column_cfg['missingValues'] = scan_yml_column.missing.values
-                    if scan_yml_column.missing.values:
-                        soda_column_cfg['missingRegex'] = scan_yml_column.missing.regex
-                    if scan_yml_column.missing.values:
-                        soda_column_cfg['missingFormat'] = scan_yml_column.missing.format
-                validity = scan_yml_column.validity
-                if validity:
-                    soda_column_cfg['validity'] = {
-                        'namedFormat': validity.format,
-                        'regexFormat': validity.regex,
-                        'validValues': validity.values,
-                        'minLength': validity.min_length,
-                        'maxLength': validity.max_length,
-                        'minValue': validity.min,
-                        'maxValue': validity.max
-                    }
-                soda_column_cfgs[column_name] = soda_column_cfg
-
-        database_and_schema = warehouse.dialect.get_warehouse_name_and_schema()
+        for column_name, scan_yml_column in scan_yml_columns.items():
+            soda_column_cfg = {}
+            if scan_yml_column.missing:
+                if scan_yml_column.missing.values:
+                    soda_column_cfg['missingValues'] = scan_yml_column.missing.values
+                if scan_yml_column.missing.values:
+                    soda_column_cfg['missingRegex'] = scan_yml_column.missing.regex
+                if scan_yml_column.missing.values:
+                    soda_column_cfg['missingFormat'] = scan_yml_column.missing.format
+            validity = scan_yml_column.validity
+            if validity:
+                soda_column_cfg['validity'] = {
+                    'namedFormat': validity.format,
+                    'regexFormat': validity.regex,
+                    'validValues': validity.values,
+                    'minLength': validity.min_length,
+                    'maxLength': validity.max_length,
+                    'minValue': validity.min,
+                    'maxValue': validity.max
+                }
+            soda_column_cfgs[column_name] = soda_column_cfg
+
         return self.execute_command({
             'type': 'sodaSqlScanStart',
-            'warehouseName': warehouse.name,
-            'warehouseType': warehouse.dialect.type,
-            'warehouseDatabaseName': database_and_schema.get('database_name'),
```

Comment:
@vijaykiran : I am not really happy with this change, the schema and database require to have a warehouse object, and some warehouses are not lazily loaded (like parsing connection details or creating a connection). This forces that the ingest has to have a working warehouse set-up, where I think ingest only should have a working Soda cloud connection, besides the dbt artifacts.